shapely/tests/test_io.py

```diff
@@ -1095,7 +1095,8 @@ def test_pickle_m(geom):
     actual = pickle.loads(pickled)
     assert_geometries_equal(actual, geom, tolerance=0)
     assert not actual.has_z
-    # TODO: assert actual.has_m
+    if not actual.is_empty:  # GEOSHasM with EMPTY geometries is inconsistent
```

Comment:
There is an upstream issue about this? 
Or is for the case of multi empty geometries (https://github.com/libgeos/geos/issues/888)?

Reply:
That's the main issue, and there are other GEOS-version-specific issues related to GEOSHasZ too like https://github.com/libgeos/geos/pull/887. These will hopefully get less complicated when support for older GEOS versions drops off. There are a few other "EMPTY geometries is inconsistent" comments that I've added to `main` with the same theme.